compiler/control/OMROptions.hpp

```diff
@@ -2517,6 +2528,13 @@ class OMR_EXTENSIBLE Options
    int32_t                     _edoRecompSizeThreshold; // Size threshold (in nodes) for candidates to recompilation through EDO
    int32_t                     _edoRecompSizeThresholdInStartupMode; // Size threshold (in nodes) for candidates to recompilation through EDO during startup
    int32_t                     _catchBlockCounterThreshold; // Counter threshold for catch blocks to trigger more aggresive inlining on the throw path
+
+   int32_t                     _arraycopyRepMovsByteArrayThreshold; // Byte array copy threshold for using REP MOVS instructions. Only supports 32 or 64 bytes
+   int32_t                     _arraycopyRepMovsCharArrayThreshold; // Char array copy threshold for using REP MOVS instructions. Only supports 32 or 64 bytes
+   int32_t                     _arraycopyRepMovsIntArrayThreshold; //  Int array copy threshold for using REP MOVS instructions. Only supports 32, 64, or 128 bytes
+   int32_t                     _arraycopyRepMovsLongArrayThreshold; // Long array copy threshold for using REP MOVS instructions. Only supports 32, 64, or 128 bytes
+   int32_t                     _arraycopyRepMovsReferenceArrayThreshold; // Reference array copy threshold for using REP MOVS instructions. Only supports 32, 64, or 128 bytes
+
```

Comment:
Because these thresholds are never validated against your stated limits, can you confirm no functional failures will occur if someone specifies a value other than the supported limits?  If the answer is that functional failures could occur then we'll need some way of validating the range of these options.

Also, since these thresholds are initialized to 0 and only changed if someone uses the option, then isn't 0 also a supported threshold?

Reply:
CG always initializes the threshold to 32 first. It sets the value from `arraycopyRepMovs*ArrayThreshold` _only_ when the value specified by `arraycopyRepMovs*ArrayThreshold`  is the supported limited and greater than 32. Otherwise, the value is ignored. There should not be any functional failure if an unsupported limit is set on the command line.

Below is an example of how a threshold is set for long array in [OMRTreeEvaluator.cpp](https://github.com/eclipse/omr/pull/7400/commits/e8065523de9adc1f64d962f8aef041b02d8238f0):

```c++
   threshold = 32;

   switch (elementSize)
      {
      case 8:
         {
         ...
         int32_t newThreshold = cg->comp()->getOptions()->getArraycopyRepMovsLongArrayThreshold();
         if ((threshold < newThreshold) && ((newThreshold == 64) || (newThreshold == 128)))
            {
            // If the CPU doesn't support AVX512, reduce the threshold to 64 bytes
            threshold = ((newThreshold == 128) && !cg->comp()->target().cpu.supportsFeature(OMR_FEATURE_X86_AVX512F)) ? 64 : newThreshold;
            }
         }
         break;
```

>since these thresholds are initialized to 0 and only changed if someone uses the option, then isn't 0 also a supported threshold? 

The initial value as `0` in the Options should be fine. It is not used to initialize the value used in CG as mentioned on the above. The value from the command line options is only considered when it is greater than 32.

Reply:
I understand that `threshold` is initialized to constant `32`, but the new options are used to initialize `newThreshold` in that code snippet.  If the options aren't explicitly set then their value of `0` is very much a part of the equation to determine if `threshold` needs to be adjusted.  That was the source of my question.

My reading of the code concludes that it should be fine though, and you verified that no functional issue will occur with an unsupported value.

Reply:
Thank you for the clarification! Should the default value for `_arraycopyRepMovs*ArrayThreshold` in OMROptions be changed to `32` as default instead of `0`?

Reply:
Yes, that seems like a sensible default if that is the minimum each of those options could be.

Reply:
Addressed in [bcf5c45](https://github.com/eclipse/omr/compare/e8065523de9adc1f64d962f8aef041b02d8238f0..bcf5c45d8ffeec198ce741dbb73a74c29a6d349d). Ready for another review. Thank you @0xdaryl